rust/src/lib.rs

```diff
@@ -2743,9 +2783,39 @@ pub unsafe extern "C" fn zcashlc_propose_shielding(
                         )
                     })
             })?
-            .keys()
-            .cloned()
-            .collect();
+            .into_iter();
+
+        let from_addrs = if let Some((receiver_if_none_provided, value)) = account_receivers.next()
```

Comment:
This is a weird way to approach this, to me. Shouldn't we be checking that the map returned by `get_transparent_balances` has length 1 and contains the transparent receiver that is provided first, and otherwise just taking the first? I don't understand the rationale for approaching the address selection in this fashion.